### PR TITLE
RouteNotFoundException when I configure a disabled module

### LINKED ISSUE #35286: RouteNotFoundException when I configure a disabled module
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

When I want to configure an activated module, no problem, the configuration page is displayed correctly.
When I disable the module I get an error, see screenshot.

image

The problem only occurs on modules using symfony controllers, on the develop branch, **no problem on v8.1**

Looking at the stack, we go to the **LegacyController**, this is new in v9. This may have something to do with the error.

For information, the module routes are loaded in the **YamlModuleLoader** class

### Expected behavior

No error when configuring disabled module

### Steps to reproduce

1. go to Modules, Module Manager
2. Disable Link List module
3. Configure Link List module

### PrestaShop version(s) where the bug happened

develop

### PHP version(s) where the bug happened

8.1

### If your bug is related to a module, specify its name and its version

ps_linklist

### Your company or customer's name goes here (if applicable).

_No response_

### PR DESCRIPTION
**Pull request by M0rgan01**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | I load the routes and services of the installed modules, not activated modules. This poses a problem when configuring a disabled module.
| Type?             | bug fix
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | CI and UI tests are green
| UI Tests          | https://github.com/M0rgan01/ga.tests.ui.pr/actions/runs/7888375959
| Fixed issue or discussion?     | Fixes #35286
| Related PRs       | -
| Sponsor company   | -

### CODE REVIEW

**Review by tleon — APPROVED** [review]

File: src/Adapter/Module/Repository/ModuleRepository.php [hunk 1]
```diff
@@ -52,6 +52,11 @@ class ModuleRepository extends AbstractObjectModelRepository
      */
     private $activeModulesPaths;
 
+    /**
+     * @var array
```
Comment (boherm) [inline src/Adapter/Module/Repository/ModuleRepository.php:56]: Not blocking for this:
```suggestion
     * @var array<string, string> File paths indexed by module name
```

**Review by boherm — APPROVED** [review]

**Review by jolelievre — APPROVED** [review]
Thanks @M0rgan01

**Comment by paulnoelcholot** [comment]
Hello @M0rgan01,

Some auto test have failed, can you check that?

Thanks!

**Comment by M0rgan01** [comment]
hello @paulnoelcholot , the UI tests are not stable, the PR https://github.com/PrestaShop/PrestaShop/pull/35307 is the fix for the failures. Everything is good for me.

**Comment by paulnoelcholot** [comment]
Hello @M0rgan01,

I tested your PR and it's good for me! :tada: 

Thanks!

**Review by paulnoelcholot — APPROVED** [review]

**Comment by mflasquin** [comment]
thanks @M0rgan01 👍

**Pull request merged**